### PR TITLE
feat(py-stepfunctions): add step-mapping to example

### PR DESCRIPTION
**Pull request by sarveshbhagat**
This example was missing step function mapping. I added that to show how to use mapping in step function using CDK (Python). Updated the image in the README.md file.

Fixes # 

---

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license].

[Apache 2.0 license]: https://www.apache.org/licenses/LICENSE-2.0

### CODE REVIEW

File: python/stepfunctions/stepfunctions/stepfunctions_stack.py [hunk 1]
```diff
@@ -9,50 +9,77 @@ class JobPollerStack(core.Stack):
     def __init__(self, app: core.App, id: str, **kwargs) -> None:
         super().__init__(app, id, **kwargs)
 
-        submit_job_activity = sfn.Activity(
```
Comment (NGL321) [inline python/stepfunctions/stepfunctions/stepfunctions_stack.py]: These don't align with our syntactic python conventions. Please revert to `xx = xx`

File: python/stepfunctions/stepfunctions/stepfunctions_stack.py [hunk 2]
```diff
@@ -9,50 +9,77 @@ class JobPollerStack(core.Stack):
...
+        ).iterator(definition_map)
+
+        definition=submit_job \
+            .next(process_map) \
+            .next(wait_x) \
+            .next(get_status) \
             .next(is_complete
                   .when(sfn.Condition.string_equals(
-                      "$.status", "FAILED"), job_failed)
+            "$.status", "FAILED"), job_failed)
```
Comment (NGL321) [inline python/stepfunctions/stepfunctions/stepfunctions_stack.py]: This is confusing. Revert to original indentation

File: python/stepfunctions/stepfunctions/stepfunctions_stack.py (hunk 1 above)
Comment (sarveshbhagat) [reply python/stepfunctions/stepfunctions/stepfunctions_stack.py]: reverted to xx = xx.

File: python/stepfunctions/stepfunctions/stepfunctions_stack.py (hunk 2 above)
Comment (sarveshbhagat) [reply python/stepfunctions/stepfunctions/stepfunctions_stack.py]: Fixed Indentation

**Comment by sarveshbhagat** [comment]
please review. Updated changes that were asked for. Thank you.

**Pull request merged**